### PR TITLE
Fixed compilation for libicu 68.

### LINKED ISSUE #47346: Make use of stdbool.h
The recent version (v68) of `libicu` deprecates its defined boolean constants for `TRUE/FALSE`: https://github.com/unicode-org/icu/blob/master/docs/userguide/dev/codingguidelines.md#primitive-types
> Traditionally, ICU4C has defined its own FALSE=0 / TRUE=1 macros for use with UBool. Starting with ICU 68 (2020q4), we no longer define these in public header files (unless U_DEFINE_FALSE_AND_TRUE=1), in order to avoid name collisions with code outside ICU defining enum constants and similar with these names.

Also see this changeset: https://github.com/unicode-org/icu/commit/c3fe7e09d8449b3d3a8deb4fc7b3bb7c97ba38ad#diff-5e55234a9864fa25c27093e7e3e3542e2ae3a4976dba37ff6ca65e8293656fab

This causes compilation errors in our globalization pal, eg.:
```
[  0%] Building CXX object pal/src/CMakeFiles/coreclrpal_dac.dir/exception/remote-unwind.cpp.o
  /home/manicka/Repositories/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_calendarData.c:129:59: error: use of undeclared identifier 'FALSE'
      GetLocale(localeName, locale, ULOC_FULLNAME_CAPACITY, FALSE, &err);
                                                            ^
```

This will cause issue only on distros with `libicu` 86 and higher, e.g.: current Manjaro 20.2.1 Nibia.

We should update our globalization pal to make use of standard `<stdbool.h>` and `true/false`.

### PR DESCRIPTION
**Pull request by ManickaP**
libicu 68 deprecated its defines for `TRUE/FALSE`. This makes use of standard `<stdbool.h>` constants.
The change only lower-cases the constant names, we were already including the header file.

Fixes https://github.com/dotnet/runtime/issues/47346

### CODE REVIEW

**Comment by tarekgh** [comment]
Looks need to do more here as I am seeing a lot of failures like the following one.

```
/__w/1/s/src/libraries/Native/Unix/System.Globalization.Native/pal_calendarData.c:129:59: error: use of undeclared identifier 'false'; did you mean 'else'?
    GetLocale(localeName, locale, ULOC_FULLNAME_CAPACITY, false, &err);

```

CC @jkotas

**Comment by jkotas** [comment]
`#include <stdbool.h>` ?

File: src/libraries/Native/Unix/System.Globalization.Native/pal_normalization.c [hunk 1]
```diff
@@ -47,7 +48,7 @@ int32_t GlobalizationNative_IsNormalized(
 
     if (U_SUCCESS(err))
     {
-        return isNormalized == TRUE ? 1 : 0;
+        return isNormalized == true ? 1 : 0;
```
Comment (jkotas) [inline src/libraries/Native/Unix/System.Globalization.Native/pal_normalization.c:51]:
```suggestion
        return isNormalized ? 1 : 0;
```

**Comment by ManickaP** [comment]
Failing tests are #47374

**Review by jkotas — APPROVED** [review]

**Comment by jkotas** [comment]
Thank you!

**Pull request merged**